### PR TITLE
ENH: Allow plugin registration sans entrypoint

### PR DESCRIPTION
**Pull request by thermokarst**
Pair-programmed with @ebolyen

### CODE REVIEW

File: qiime2/sdk/plugin_manager.py [hunk 1]
```diff
@@ -38,29 +38,30 @@ def iter_entry_points(cls):
...
+            # These are all dependent loops, each requires the loop above it to
+            # be completed.
+            for entry_point in self.iter_entry_points():
+                plugin = entry_point.load()
+                self.plugins[plugin.name] = plugin
 
-        for plugin in self.plugins.values():
-            self._integrate_plugin(plugin)
+            for plugin in self.plugins.values():
+                self._integrate_plugin(plugin)
```
Comment (ebolyen) [inline qiime2/sdk/plugin_manager.py]: So it has occurred to me that this loop might be happening after setting up the plugin dict on purpose, which `install_plugin` can't do without a `finalize()` method of some kind.

**Comment by ebolyen** [comment]
Looking at this again, I don't see any reason this won't work. Any dependence between plugin namespaces is checked in `_integrate_plugin` which presumes previous plugins have already run through `_integrate_plugin`. So the difference between updating the plugin dict and then integrating per-plugin, and filling the plugins dict and then integrating over all shouldn't matter.

**Comment by thermokarst** [comment]
Here is an example invocation, @Oddant1:

https://github.com/thermokarst/q2galaxy/blob/dummy/q2galaxy/__init__.py#L59-L64

**Comment by thermokarst** [comment]
Thanks for looking into this, @Oddant1! This PR has been superseded by #458.

**Pull request closed without merging**